src/unwind_dwfl.cc

```diff
@@ -87,7 +88,35 @@ static void trace_unwinding_end(UnwindState *us) {
 static DDRes add_dwfl_frame(UnwindState *us, const Dso &dso, ElfAddress_t pc,
                             DDProfMod *ddprof_mod, FileInfoId_t file_info_id);
 
-// returns true if we should continue unwinding
+
+
+// check for runtime symbols provided in /tmp files
+static DDRes add_runtime_symbol_frame(UnwindState *us, const Dso &dso,
+                                      ElfAddress_t pc) {
+  SymbolHdr &unwind_symbol_hdr = us->symbol_hdr;
+  SymbolTable &symbol_table = unwind_symbol_hdr._symbol_table;
+  RuntimeSymbolLookup &runtime_symbol_lookup =
+      unwind_symbol_hdr._runtime_symbol_lookup;
+  SymbolIdx_t symbol_idx =
+      runtime_symbol_lookup.get_or_insert(dso._pid, pc, symbol_table);
+  if (symbol_idx == -1) {
+    add_dso_frame(us, dso, pc, "pc");
+    return ddres_init();
+  }
+  UnwindOutput *output = &us->output;
+  int64_t current_loc_idx = output->nb_locs;
```

Comment:
Check for `UnwindOutput::locs` overflow is missing, the code for adding a new location could probably be factorized.